posthog/caching/warming.py

```diff
@@ -37,16 +40,48 @@
 LAST_VIEWED_THRESHOLD = timedelta(days=7)
 
 
-def priority_insights(team: Team, shared_only: bool = False) -> Generator[tuple[int, Optional[int]], None, None]:
+def teams_enabled_for_cache_warming() -> list[int]:
+    all_teams = Team.objects.all()
```

Comment:
this will OOM. It sounds like you only need the `organization_id` and `team.id` so fetch only those things from the DB. There's a lot of extra stuff in there.

best bet here is to follow what the usage report does: yes iterate across all teams, but do it via a generator vs. loading them all in memory first and also potentially chunk if you need to future proof this.

Reply:
the really annoying thing about this happening in celery is its hard to debug, because when it OOMs it just dies and the task fails, that's all we know _sigh_.

Reply:
Sounds good, was gonna do just that but decided against it after trying it out in the prod shell. Though you're right, might as well do it to future proof it, don't want to debug celery jobs in prod 😅